### ISSUE TITLE
Maya To Hydra (MtoH) built with PixarUSD 21.08 or later: Selection issue on hdRPR and Arnold render delegates

### ISSUE THREAD

**Issue by Vsevolod1983** [Autodesk/maya-usd#1741]
**Describe the bug**
Hello !
I use Pixar USD within Maya To Hydra (MtoH) project for rendering using hdRPR render delegate.
Actually the problem I'm going to share also gets reproduced on Arnold render delegate as well.

I have strange issue regarding selection. When I select any object in Maya, e.g. simple cube, all rendered picture suddenly disappears in the viewport (for hdRPR or Arnold Hydra render delegate).
I made some investigation and figured out that PixarUSD 21.05 works ok, problem started to appear on PixarUSD 21.08

Some investigation regarding commits (inside PixarUSD repo)
Pixar USD 21.05 (090ef0d) - works ok, selected mesh is fully filled with yellow color.
![image](https://user-images.githubusercontent.com/7079246/136180720-da6d819b-a2ad-4ac8-950a-dfaa9380495d.png)

Starting from commit Apr 15
c1d846c ([Hgi] Added buffer binding support for Hgi shader codegen) - selection works but selected mesh looks like just outlined (not fully filled with yellow color).
![image](https://user-images.githubusercontent.com/7079246/136172303-cfa7fa6f-a5d1-45ee-b6d9-4d96de2d4d20.png)

commit Apr 27
36223b8 (Do not restore framebuffer in HgiGLScopedStateHolder and HgiInterOpOpenGL.) - selection does not work at all. If any of meshes is selected, the entire rendered image gets disappeared.
![image](https://user-images.githubusercontent.com/7079246/136172492-4773e33a-300c-418e-aaa2-1fc3ec8ab897.png)

If disable execution of ColorizeSelectionTask in Hydra - image does not get disappeared but obviously selection does not work.

I see no issues while using the same render delegate hdRPR in USD Viewer application with any version of PixarUSD (21.05, 21.08, 21.11). 

So it looks like something wrong with the way how Maya viewport interacts with graphics low level of PixarUSD.
All commits I mentioned above have changes in things regarding textures, GL calls, etc.

Does anybody could help with that ? (PixarUSD guys redirected me to this repository for posting an issue).
Thank you!

**Steps to reproduce**
1. Use MtoH plugin with PixarUSD 21.08 or later
2. Select Hydra Render Delegate (Use Arnold render delegate for example)
3. Make a cube and select it
4. Notice that the entire image gets disappeared.

**Expected behavior**
If you select a mesh entire scene should not get disappeared.

**Specs (if applicable):**
 - Windows 10
 - Visual Studio 2017 toolset 141
 - Maya 2022.2
 - Could be reproduced on any version of MtoH.
 - Pixar USD commits are listed in issue's description

**Comment by kxl-adsk** [comment]
@Vsevolod1983 we have a PR #1863 open to address this issue. Feel free to test it out before we merge.

**Comment by Vsevolod1983** [comment]
I will try, thank you !

**Comment by Vsevolod1983** [comment]
@kxl-adsk 
Hello !
Thank you for the fix!
The image does not disappear anymore, however, selection looks like just being outlined (selected shape is not fully-filled with yellow color).
Is it expected ?

![image](https://user-images.githubusercontent.com/7079246/144267540-c41a4c58-e995-4dd9-9186-6099220519e3.png)

**Comment by perrauo-adsk** [comment]
Yes, the image disappearing was tied to fc63eaef295edb2f90099614f833c26fac11764c.

We will have a look at the outline issue separately. I am not sure of the root cause yet.

**Comment by williamkrick** [comment]
Hi @Vsevolod1983, I took a look at this one today. In the `Arnold (Hydra)` render settings the default value for `Selection Outline` (cut off in my image) is 4.0, which turns on selection outline for selection highlighting. In `GL (Hydra)` the default value is 0 and so the objects draw with full highlight.

Is it possible that you have the default outline value set for `Arnold (Hydra)` and that is why you are seeing outlines? Locally with USD 21.11 & up-to-date MayaUSD when I set the outline to 0 pixels I see objects with full highlighting.

![image](https://user-images.githubusercontent.com/24392049/148808895-dab6bc88-32b2-4043-afba-330324f081a8.png)
![image](https://user-images.githubusercontent.com/24392049/148809225-6cd64219-24b4-4a7b-9b76-b319853d45e1.png)
